tracer/src/Datadog.Trace/Ci/Test.cs

```diff
@@ -211,17 +211,73 @@ public void SetTestMethodInfo(MethodInfo methodInfo)
+            }
+        }
+    }
+
+    internal static void SetStringOrArray(TestSpanTags testTags, TestSuiteSpanTags suiteTags, Func<TestSpanTags, string?> getTestTag, Func<TestSuiteSpanTags, string?> getSuiteTag, Action<TestSuiteSpanTags, string?> setSuiteTag)
+    {
+        // If the value is not set, we set it to the current test tag
+        // If it is set, we check if it is an array and add the current test tag to it
+        // If it is not an array, we create a new array with both values
+        // This is to support multiple values in a single tag
+        var suiteTagValue = getSuiteTag(suiteTags);
+        var testTagValue = getTestTag(testTags);
+        if (StringUtil.IsNullOrEmpty(testTagValue))
+        {
+            return;
+        }
+
+        if (StringUtil.IsNullOrEmpty(suiteTagValue))
+        {
+            setSuiteTag(suiteTags, testTagValue);
+        }
+        else if (!string.Equals(suiteTagValue, testTagValue, StringComparison.OrdinalIgnoreCase))
+        {
+            if (suiteTagValue.StartsWith("[", StringComparison.OrdinalIgnoreCase) &&
+                suiteTagValue.EndsWith("]", StringComparison.OrdinalIgnoreCase))
             {
-                var match = codeOwners.Match("/" + ciValues.MakeRelativePathFromSourceRoot(methodSymbol.File, false));
-                if (match is not null)
+                // If the source file is an array, we add the new source file to it
+                var files = Vendors.Newtonsoft.Json.JsonConvert.DeserializeObject<List<string>>(suiteTagValue);
+                if (files is not null && !files.Contains(testTagValue))
                 {
-                    tags.CodeOwners = "[\"" + string.Join("\",\"", match) + "\"]";
+                    files.Add(testTagValue);
+                    setSuiteTag(suiteTags, Vendors.Newtonsoft.Json.JsonConvert.SerializeObject(files));
                 }
             }
+            else
+            {
+                // If the source file is not an array, we create a new one with both values
+                setSuiteTag(suiteTags, Vendors.Newtonsoft.Json.JsonConvert.SerializeObject(new List<string> { suiteTagValue, testTagValue }));
+            }
+        }
+    }
+
+    internal static void SetCodeOwnersOnTags(TestSpanTags testTags, TestSuiteSpanTags suiteTags, IEnumerable<string> codeOwners)
+    {
+        testTags.CodeOwners = "[\"" + string.Join("\",\"", codeOwners) + "\"]";
+        if (StringUtil.IsNullOrEmpty(suiteTags.CodeOwners))
+        {
+            suiteTags.CodeOwners = testTags.CodeOwners;
+        }
+        else if (!string.Equals(suiteTags.CodeOwners, testTags.CodeOwners, StringComparison.OrdinalIgnoreCase))
+        {
+            var suiteCodeOwners = Vendors.Newtonsoft.Json.JsonConvert.DeserializeObject<List<string>>(suiteTags.CodeOwners) ?? [];
+            suiteCodeOwners.AddRange(codeOwners);
+            suiteTags.CodeOwners = "[\"" + string.Join("\",\"", suiteCodeOwners.Distinct()) + "\"]";
```

Comment:
Distinct probably needs to check `OrdinalIgnoreCase` for consistency?

Reply:
done in [0167a01](https://github.com/DataDog/dd-trace-dotnet/pull/7295/commits/0167a01cdb55d2d84639d680ffc7a51c9e837af9)